src/pop/blockchain/alt_block_tree.cpp

```diff
@@ -568,6 +571,27 @@ bool AltBlockTree::setState(index_t& to, ValidationState& state) {
   return success;
 }
 
+void AltBlockTree::doFinalize() {
+  ValidationState state;
+  auto* tip = getBestChain().tip();
+  VBK_ASSERT(tip && "must be bootstrapped");
+
+  int32_t maxReorg = (int32_t)getParams().getMaxReorgDistance();
+  if (maxReorg > getRoot().getHeight()) {
+    return;
+  }
```

Comment:
It looks like finalization won't work until root reaches maxReorg height, eg if maxReorg is 100000 finalization won't work until root is on height 100000.

Reply:
you're right, it's a typo. we should compare maxReorg with tip height, not root height